src/content/blog/2023_qlasskit.md

```diff
@@ -0,0 +1,149 @@
+---
+title: Qlasskit - A bridge between Python and quantum algorithms
+author: Davide Gessa
+day: 13
+month: 12
+year: 2023
+---
+
+Traditionally, creating quantum circuits requires specialized knowledge in quantum programming. This requirement holds true when encoding a classical algorithm inside a quantum circuit, for instance, for an oracle or a black-box component of a quantum algorithm. This often becomes a time wasting job, since we almost always already have a classical implementation in a traditional high level language.
+
+**Qlasskit**, an open-source Python library developed with the support of a Unitary Fund microgrant, addresses this challenge head-on by allowing direct translation of standard Python code into invertible quantum circuits without any modification to the original code. Furthermore, *qlasskit* implements some well-known quantum algorithms and offers a comprehensive interface for implementing new ones.
+
+*Qlasskit* adopts a distinctive method where it constructs a *single boolean expression* for each output qubit of the entire function, rather than translating individual operations into quantum circuits and then combining them. This approach enables advanced optimization by leveraging boolean algebraic properties.
+
+For instance, let assume we have the following function:
+
+```python
+from qlasskit import qlassf, Qint2, Qint4
+from qiskit import QuantumCircuit
+
+@qlassf
+def f_comp(b: bool, n: Qint2) -> Qint2:
+      for i in range(3):
+            n += (1 if b else 2)
+      return n
+```
+
+The first things you can notice in this code are:
+
+- the `qlassf` decorators, indicating that the function will be translated to a quantum circuit.
+- special bit-sized types `Qint4`, and `Qint8`. These are required as qubits are a precious resource, and we want to use as few as possible.
+- and, it is normal Python code.
```

Comment:
Suggestion: "- it all reads as normal Python code, i.e., no quantum routine is visible."

Reply:
Ok, modified; the second part of your sentence is not true, since qlasskit allows certain types of classical-quantum hybrids (still experimental feature indeed)